src/updater/csv_source.cpp

```diff
@@ -0,0 +1,47 @@
+#include "updater/csv_source.hpp"
+
+#include "updater/csv_file_parser.hpp"
+
+#include <boost/fusion/adapted/std_pair.hpp>
+#include <boost/fusion/include/adapt_adt.hpp>
+
+BOOST_FUSION_ADAPT_STRUCT(osrm::updater::Segment,
+                          (decltype(osrm::updater::Segment::from),
+                           from)(decltype(osrm::updater::Segment::to), to))
+BOOST_FUSION_ADAPT_STRUCT(osrm::updater::SpeedSource,
+                          (decltype(osrm::updater::SpeedSource::speed),
+                           speed)(decltype(osrm::updater::SpeedSource::weight), weight))
+BOOST_FUSION_ADAPT_STRUCT(osrm::updater::Turn,
+                          (decltype(osrm::updater::Turn::from),
+                           from)(decltype(osrm::updater::Turn::via),
+                                 via)(decltype(osrm::updater::Turn::to), to))
+BOOST_FUSION_ADAPT_STRUCT(osrm::updater::PenaltySource,
+                          (decltype(osrm::updater::PenaltySource::duration),
+                           duration)(decltype(osrm::updater::PenaltySource::weight), weight))
```

Comment:
urgh format seems to be broken here, what about

    // clang-format off
    manually format
    // clang-format on